tools/fast/fastesting/basic.go

```diff
@@ -59,9 +62,12 @@ func NewTestEnvironment(ctx context.Context, t *testing.T, fastenvOpts fast.File
 	genesisMiner, err := env.GenesisMiner()
 	require.NoError(t, err)
 
-	fastenvOpts = fast.FilecoinOpts{
-		InitOpts:   append([]fast.ProcessInitOption{fast.POGenesisFile(genesisURI)}, fastenvOpts.InitOpts...),
-		DaemonOpts: append([]fast.ProcessDaemonOption{fast.POBlockTime(100 * time.Millisecond)}, fastenvOpts.DaemonOpts...),
+	fastenvOpts.InitOpts = append([]fast.ProcessInitOption{fast.POGenesisFile(genesisURI)}, fastenvOpts.InitOpts...)
+
+	// The old method appended a default blocktime and this caused a bug when callers set blocktime
+	// as well. See Issue
+	if len(fastenvOpts.DaemonOpts) == 0 {
```

Comment:
Should we instead explicitly check if the `POBlockTime` opt was passed here? There could be daemon opts passed that are unrelated to block time.

Reply:
do you know how to do that? I couldn't figure out how

Reply:
I don't think there is a super clean way to actually check if the blocktime was passed. I guess the best thing to do would be to iterator over all `DaemonOpts` invoking them and doing a check for `--block-time`.

This might be the best way to move forward at the moment.

```
isBlockTimeSet := false
for _, fn := range fastenvOpts.DaemonOpts {
	s := fn()
	if len(s) > 0 && s[0] == "--block-time" {
		isBlockTimeSet = true
		break
	}
}
```

Currently the requirement to change the blocktime stems from an issue of having to use `MiningStart` to get around an issue where the storage protocol and sector builder is tied to mining. I have an issue to track this in #2968.

So I'm not sure how common it's going to be to actually want to change the blocktime after we get a fix in. The issue we face is that the actual default of 30s is not what we want to default to in these tests most of the time. So I think regardless we are going to want to be able to detect block-time being passed, and add the daemon opt if it's not provided.

Reply:
Ok so should this change be isolated to detecting if a block time was passed in and then adding one if it's not?

Reply:
Ya, I think everything else in this PR looks good.